Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp

```diff
@@ -552,7 +552,11 @@ static gboolean wpeViewWaylandRenderBuffer(WPEView* view, WPEBuffer* buffer, GEr
 
     auto* wlSurface = wpe_view_wayland_get_wl_surface(WPE_VIEW_WAYLAND(view));
     wl_surface_attach(wlSurface, wlBuffer, 0, 0);
-    wl_surface_damage(wlSurface, 0, 0, wpe_view_get_width(view), wpe_view_get_height(view));
+    if (damageRects) {
+        for (unsigned i = 0; i < damageRectsCount; ++i)
+            wl_surface_damage(wlSurface, damageRects[i].x, damageRects[i].y, damageRects[i].width, damageRects[i].height);
```

Comment:
I think we should use here `wl_surface_damage_buffer()` instead, because the damage rectangles are in physical pixel sizes, right? There is a [note about this](https://wayland-book.com/surfaces-in-depth/damaging-surfaces.html) in the Wayland book:

> The Wayland protocol provides two requests for damaging surfaces: `damage` and `damage_buffer`. The former is effectively deprecated, and you should only use the latter. The difference between them is that damage takes into account all of the transforms affecting the surface, such as rotations, scale factor, and buffer position and clipping. The latter instead applies damage relative to the buffer, which is generally easier to reason about.

Reply:
Our API documentation should always indicate whether sizes are physical or logical, by the way. Which is not done right now and it's impossible to know e.g. if `wpe_view_get_{width,height}()` use physical or logical (taking into account the scaling factor).

Reply:
Okay, I figured this out while working on this: sizes in the damage rects are physical pixels (the scaling factor is _not_ taken into account).